### PR TITLE
added Gcovr coverage tool support

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

File: CoverageGcovr.cmake [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+if(ENABLE_GCOVR)
+  if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_COMPILER_IS_CLANG)
+     find_program(GCOVR gcovr)
+     if(NOT GCOVR)
+       message(FATAL_ERROR "No code coverage report, ${COVERAGE_MISSING}")
+     endif()
+
+     set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${__profiling_flags}")
+     set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} ${__profiling_flags}")
```
Comment (eile) [inline CoverageGcovr.cmake]: Only in debug builds?

**Comment by juanchopanza** [comment]
Is there something else that ensures that coverage only happens in DEBUG build type? Otherwise shouldn't this modify more general variables (possibly `CMAKE_CXX_FLAGS` etc. although I am not sure how this gets handled here.)

**Comment by eile** [comment]
Is there a reason to limit coverage to debug builds?

**Comment by juanchopanza** [comment]
Probably for reliability with `gcov`, coverage stats should be obtained without optimization. See

https://gcc.gnu.org/onlinedocs/gcc/Gcov-and-Optimization.html

**Comment by eile** [comment]
Ok, then looks good for me. @mgeplf ready to merge?

**Comment by mgeplf** [comment]
Fine w/ me

**Pull request merged**